src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketMirrorHandler.java

```diff
@@ -0,0 +1,125 @@
+package com.atlassian.bitbucket.jenkins.internal.scm;
+
+import com.atlassian.bitbucket.jenkins.internal.client.BitbucketClientFactory;
+import com.atlassian.bitbucket.jenkins.internal.client.BitbucketClientFactoryProvider;
+import com.atlassian.bitbucket.jenkins.internal.client.BitbucketMirrorClient;
+import com.atlassian.bitbucket.jenkins.internal.client.exception.BitbucketClientException;
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketPluginConfiguration;
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketServerConfiguration;
+import com.atlassian.bitbucket.jenkins.internal.credentials.BitbucketCredentials;
+import com.atlassian.bitbucket.jenkins.internal.credentials.BitbucketCredentialsAdaptor;
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketMirroredRepository;
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketMirroredRepositoryDescriptor;
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketMirroredRepositoryStatus;
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketRepository;
+import com.cloudbees.plugins.credentials.common.StandardListBoxModel;
+import hudson.util.ListBoxModel.Option;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.emptyMap;
+import static java.util.Objects.requireNonNull;
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+class BitbucketMirrorHandler {
```

Comment:
Both methods in this class are only used once, both times in `BitbucketSCM`. Would it be better to move them into `BitbucketSCM` as private methods?
It especially feels weird that this class returns both `StandardListBoxModel` and `EnrichedBitbucketMirroredRepository`, two very different types of objects

Reply:
It also might mean you can get rid of `EnrichedBitbucketMirroredRepository` entirely

Reply:
The problem with moving things in BitbucketSCM is, 
1. Difficulty in unit testing Mirrors handling which is really easy
2. Club all mirror handling things together so it is more cohesive and future changes are contained. 
I can keep StandardListBoxModel in BitbucketSCM but then that would split the mirrors things between BitbucketSCM and this class. I agree with these 2 methods looks odd together but it is a trade-off. 

> Both methods in this class are only used once
That's why I have added them in the same package and will be package default. 

Another important benefit is the ability to have an integration test between mirror handling and bitbucket server without Jenkins.